File: minimal.py
import sys

pairs = []
values = {}
k = int(sys.argv[1])

for line in sys.stdin:
    f,s = map(int,line.split('|'))
    values.setdefault(s,set()).add(f)
    pairs.append((f,s))

for a,b in pairs:
    if len(values[b]) >= k:
        print "%d|%d" % (a,b)

File: minimal.rb
k=ARGV[0].to_i
require 'set'
pairs = $stdin.map {|line| x,y = line.chop.split('|')}
vals = {}
pairs.each{ |x,y| vals[y.to_i] = (vals[y.to_i] || Set.new).add(x.to_i) }
pairs.each{ |x,y| if(vals[y.to_i].size() >= k) then printf "%d|%d\n",x,y end }
